### PR TITLE
feat: copilot new api VSCODE-550

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Create a MongoDB participant using the latest VSCode [Language Model API](https://code.visualstudio.com/api/extension-guides/language-model)
- Create the `/query` command that generated MongoDB queries and can open them in a new playground or run directly
- Call a generic handler if the participant called without a command

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/participant/participant.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+    stream.button({
+      command: EXTENSION_COMMANDS.RUN_PARTICIPANT_QUERY,
+      title: vscode.l10n.t('▶️ Run'),
+    });
+
+    stream.button({
+      command: EXTENSION_COMMANDS.OPEN_PARTICIPANT_QUERY_IN_PLAYGROUND,
+      title: vscode.l10n.t('Open in playground'),
+    });
```
Comment (Anemy) [inline src/participant/participant.ts:206]: Maybe not for this pr, but I think there is a chance the ai could return multiple code blocks in their response. 
Like the prompt:
> What are 3 ways I could do X

I'm thinking we probably want to add a play and open in playground button for each code block as they come. Ideally we show them inline, right after the code block closes, while we stream the response.

File: src/editors/playgroundController.ts [hunk 2]
```diff
@@ -164,6 +169,7 @@ export default class PlaygroundController {
     this._exportToLanguageCodeLensProvider = exportToLanguageCodeLensProvider;
     this._playgroundSelectedCodeActionProvider =
       playgroundSelectedCodeActionProvider;
+    this._participantController = participantController;
```
Comment (Anemy) [inline src/editors/playgroundController.ts:172]: Why do we have this on the playgroundController class? Maybe we can remove it?

**Review by Anemy — APPROVED** [review]
lgtm, left one small question and suggestion. I'm thinking we'll be doing a good amount of iterating on prompts,  response handling, and more so not a need for now.

File: src/editors/playgroundController.ts (hunk 2 above)
Comment (alenakhineika) [reply src/editors/playgroundController.ts:172]: Planned for the feature usage, but I can remove it for now. As you wish. I just fixed tests and didn't want to revert this work 🙈

File: src/participant/participant.ts (hunk 1 above)
Comment (alenakhineika) [reply src/participant/participant.ts:206]: Good point. Let's address this in further PRs 👍

File: src/editors/playgroundController.ts (hunk 2 above)
Comment (Anemy) [reply src/editors/playgroundController.ts:172]: All good ! No worries!

**Pull request merged**